### PR TITLE
Task/svgiconprovider

### PR DESCRIPTION
**Pull request by ochorocho**
Sorry, messed it up.

This is the origin PR: https://github.com/s2b/vite-asset-collector/pull/9

### CODE REVIEW

File: Classes/IconProvider/SvgIconProvider.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+     * @throws \InvalidArgumentException|ViteException|ExtensionConfigurationExtensionNotConfiguredException|ExtensionConfigurationPathDoesNotExistException
+     */
+    protected function generateInlineMarkup(array $options): string
+    {
+        if (empty($options['source'])) {
+            throw new \InvalidArgumentException('The option "source" is required and must not be empty', 1690831431);
+        }
+
+        $source = $this->viteService->getAssetPathFromManifest(
+            $options['manifest'] ?? $this->extensionConfiguration->get('vite_asset_collector', 'defaultManifest'),
```
Comment (s2b) [inline Classes/IconProvider/SvgIconProvider.php:68]: here you can re-use `getManifest()`

File: README.md [hunk 2]
```diff
@@ -257,3 +257,17 @@ editor:
             # Using another manifest.json
             - "%vite('EXT:sitepackage/Resources/Private/Css/Rte.css', 'path/to/manifest.json')%"
 ```
+
+### IconProvider
+
+To use an icon along with `core:icon` add the following to the `Icons.php` file:
```
Comment (s2b) [inline README.md:263]: I think that there needs to be a bit more documentation:

* When and why would I want to use this?
* Which adjustments need to be made to vite.config.js?
* Where is this applicable? BE and/or FE?
* A link to [the documentation](https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Icon/Index.html)

Comment (s2b) [reply README.md:263]: I can also add this myself after merging.

**Review by s2b — CHANGES_REQUESTED** [review]

File: Classes/IconProvider/SvgIconProvider.php (hunk 1 above)
Comment (ochorocho) [reply Classes/IconProvider/SvgIconProvider.php]: Done :-)

**Review by s2b — APPROVED** [review]

**Pull request merged**